msu/systems/mod_settings/settings_element.nut

```diff
@@ -0,0 +1,109 @@
+::MSU.Class.SettingsElement <- class
+{
+	Name = null;
+	ID = null;
+	static Type = "Element";
+	Flags = null;
+	Description = null;
+	Parent = null;
+
+	constructor(_id, _name = null)
+	{
+		if (_id.find(" ") != null)
+		{
+			::logError("The ID of a Settings Element should not have spaces");
+			throw ::MSU.Exception.InvalidType;
+		}
+		this.Name = _name == null ? _id : _name;
+		this.ID = _id;
+		this.Flags = [];
+		this.Description = "";
+	}
+
+	function getType()
+	{
+		return this.Type;
+	}
+
+	function setParent( _parent )
+	{
+		this.Parent = _parent.weakref();
+	}
+
+	function getParent()
+	{
+		return this.Parent;
+	}
+
+	function getPanelID()
+	{
+		return this.getParent().getParent().getID();
+	}
+
+	function setDescription( _description )
+	{
+		this.Description = _description;
+	}
+
+	function getDescription()
+	{
+		return this.Description;
+	}
+
+	function getName()
+	{
+		return this.Name;
+	}
+
+	function getID()
+	{
+		return this.ID;
+	}
+
+	function addFlags( ... )
```

Comment:
Shouldn't this instead accept an array as argument?

Reply:
I don't see why